### PR TITLE
refactor join cluster process

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. join cluster after start all service, this can make K8S readiness probe happy, thus the node host name can be resolved by DNS.
2. use node id as node name in add_node,instead of an empty string.

## Changelog

- Bug Fix

## Related Issues

Fixes   [4218](https://github.com/datafuselabs/databend/issues/4218)

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: metasrv/src/meta_service/raftmeta.rs [hunk 1]
```diff
@@ -412,6 +412,47 @@ impl MetaNode {
         Ok(mn)
     }
 
+    pub async fn join_cluster(&self, conf: &RaftConfig) -> MetaResult<()> {
+        if conf.join.is_empty() {
+            return Ok(());
```
Comment (ariesdevil) [inline metasrv/src/meta_service/raftmeta.rs:417]: Should we log a warning message when this config is empty?

File: metasrv/src/meta_service/raftmeta.rs [hunk 2]
```diff
@@ -412,6 +412,47 @@ impl MetaNode {
...
+        // Try to join a cluster only when this node is just created.
+        // Joining a node with log has risk messing up the data in this node and in the target cluster.
+        if self.is_opened() {
+            return Ok(());
+        }
+
+        let addrs = &conf.join;
+        // Join cluster use advertise host instead of listen host
+        let raft_api_advertise_host_endpoint = conf.raft_api_advertise_host_endpoint();
+        #[allow(clippy::never_loop)]
```
Comment (ariesdevil) [inline metasrv/src/meta_service/raftmeta.rs:429]: Why not just use `addrs[0]` ?

File: metasrv/src/meta_service/raftmeta.rs (hunk 1 above)
Comment (drmingdrmer) [reply metasrv/src/meta_service/raftmeta.rs:417]: I think the elaborated name of this method should be `try_to_join_cluster_if_cli_arg_join_is_specified()`.

`conf.join` being empty means the user did not pass `--join` arguments to metasrv. It's totally OK.

File: metasrv/src/meta_service/raftmeta.rs (hunk 2 above)
Comment (drmingdrmer) [reply metasrv/src/meta_service/raftmeta.rs:429]: It's an unfinished code snippet... There should be a retry loop... :((

**Review by drmingdrmer — APPROVED** [review]
Looks pretty good to me:DDD

**Review by ZhiHanZ — APPROVED** [review]
tested and works, thanks @lichuang

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**